internal/controller/managedcluster_controller.go

```diff
@@ -125,14 +129,19 @@ func (r *ManagedClusterReconciler) setStatusFromClusterStatus(
 
 	allConditionsComplete := true
 	for _, metaCondition := range resourceConditions.Conditions {
-		if metaCondition.Status != "True" {
-			allConditionsComplete = false
-		}
+		userFriendlyTypeName, ok := obj.statuses[metaCondition.Type]
```

Comment:
I don't get it, why do we need this extra field, already having human-readable `kind` and its `name`? And do we need the `requeueIfNoObjects` field which is unused? So the whole code may be simplified to just the following method alone:

```go
func (r *ManagedClusterReconciler) setStatusFromChildObjects(ctx context.Context, managedCluster *hmc.ManagedCluster) (requeue bool, merr error) {
	l := ctrl.LoggerFrom(ctx)

	allConditionsComplete := true
	for _, gvr := range []schema.GroupVersionResource{
		{
			Group:    "cluster.x-k8s.io",
			Version:  "v1beta1",
			Resource: "clusters",
		},
		{
			Group:    "cluster.x-k8s.io",
			Version:  "v1beta1",
			Resource: "machinedeployments",
		},
	} {
		resourceConditions, err := status.GetResourceConditions(ctx, managedCluster.Namespace, r.DynamicClient, gvr,
			labels.SelectorFromSet(map[string]string{hmc.FluxHelmChartNameKey: managedCluster.Name}).String())
		if err != nil {
			if errors.As(err, &status.ResourceNotFoundError{}) {
				l.Info(err.Error())
				allConditionsComplete = false
				continue
			}

			merr = errors.Join(merr, fmt.Errorf("failed to get conditions: %w", err))
			continue
		}

		for _, metaCondition := range resourceConditions.Conditions {
			if metaCondition.Status != "True" {
				metaCondition.Message = resourceConditions.Kind + ":" + metaCondition.Message
				allConditionsComplete = false
			}

			if metaCondition.Reason == "" && metaCondition.Status == "True" {
				metaCondition.Message = resourceConditions.Kind + " is Ready"
				metaCondition.Reason = hmc.SucceededReason
			}

			apimeta.SetStatusCondition(managedCluster.GetConditions(), metaCondition)
		}
	}

	return !allConditionsComplete, merr
}
```

Reply:
requeueIfNoObjects is a leftover from the previous version of code (before merge) where we had no status.GetResourceConditions and logic was a bit different.
Thanks for the great comments!